homeassistant/components/blink/config_flow.py

```diff
@@ -58,7 +59,7 @@ def validate_input(hass: core.HomeAssistant, auth):
         raise Require2FA
 
 
-def _send_blink_2fa_pin(auth, pin):
+def _send_blink_2fa_pin(auth: Auth, pin: str) -> bool:
```

Comment:
I believe the executor is hiding this problem.

https://github.com/home-assistant/core/blob/cbbdb2b545249ece78b4a71923ebc76877f8eb3c/homeassistant/components/blink/config_flow.py#L125-L129

The pin here is `Any|None` and should be converted to a string?
 'pin: str = str(user_input.get(CONF_PIN))'

Reply:
I just looked at the code and guessed what it should be. The pin comes from user input right? (Not behind a computer atm)

Reply:
Yes, it comes from user input (which is `Any/None` return type.  I only caught it in my PR because I removed the executor (moving to async for the call) and mypy complained.  I think a string is what we want, just need to cast it with `str()`, but I'm still running with training wheels :)

Reply:
Ah right. Hmm, well at this point Mypy doesn't complain and currently it's handled as a string, so I think we're good in this PR. Do you agree?

Reply:
Sure, it's "ok" as it is passing (even though it's not technically "right"), but it will end up in my PR (which I was hoping to make "smaller") because it's not "right".  Either way it will be addressed.

Reply:
I think currently it's technically right as we know that `user_input[CONF_PIN]` is `str` and not `Any` (since we defined our schema like that). And with this change, we still treat it as a `str`.

But potato potato, in the end we will make sure that it's flushed and fine :)